content/_code-samples/build-a-wallet/desktop-js/library/4_helpers.js

```diff
@@ -0,0 +1,27 @@
+const xrpl = require("xrpl");
+
+const prepareTxData = (transactions) => {
+    return transactions.map(transaction => ({
+        confirmed: transaction.tx.date,
+        type: transaction.tx.TransactionType,
+        from: transaction.tx.Account,
+        to: transaction.tx.Destination,
+        value: getDisplayableAmount(transaction.meta.delivered_amount),
+        hash: transaction.tx.hash
+    }))
+}
+
+const getDisplayableAmount = (rawAmount) => {
+    if (rawAmount === 'unavailable') {
```

Comment:
Similar to above, not all transactions have a `delivered_amount` field. (There are ones that don't deliver value, like AccountSet for example, and ones that [don't have it implemented](https://github.com/XRPLF/rippled/issues/3492) like EscrowFinish.)

The browser wallet handles this in the code that calls this function, but for this tutorial I think it's easier to handle it here by adding a case for `undefined`:

```suggestion
    if (rawAmount === undefined) {
        // Not all transaction types have delivered_amount
        return "-"
    } else if (rawAmount === 'unavailable') {
```